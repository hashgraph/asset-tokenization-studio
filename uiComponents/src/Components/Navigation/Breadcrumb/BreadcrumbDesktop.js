var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __spreadArray = (this && this.__spreadArray) || function (to, from, pack) {
    if (pack || arguments.length === 2) for (var i = 0, l = from.length, ar; i < l; i++) {
        if (ar || !(i in from)) {
            if (!ar) ar = Array.prototype.slice.call(from, 0, i);
            ar[i] = from[i];
        }
    }
    return to.concat(ar || Array.prototype.slice.call(from));
};
import React from "react";
import { Breadcrumb as ChakraBreadcrumb, BreadcrumbItem as ChakraBreadcrumbItem, BreadcrumbLink as ChakraBreadcrumbLink, Link as ChakraLink, Menu, MenuButton, Skeleton as ChakraSkeleton, } from "@chakra-ui/react";
import { forwardRef, useMultiStyleConfig as useChakraMultiStyleConfig, } from "@chakra-ui/system";
import { PhosphorIcon } from "../../Foundations/PhosphorIcon";
import { omit as _omit } from "lodash";
import { CaretRight, DotsThree } from "@phosphor-icons/react";
import { Dropdown, DropdownItem } from "@Components/DataDisplay/Dropdown";
var DEFAULT_MENU_BUTTON_LABEL = "...";
var HiddenItems = function (_a) {
    var items = _a.items;
    var menu = useChakraMultiStyleConfig("Breadcrumb").menu;
    return (React.createElement(Menu, { "data-testid": "breadcrumb-menu" },
        React.createElement(MenuButton, { "data-testid": "breadcrumb-menu-button", sx: menu },
            React.createElement(PhosphorIcon, { as: DotsThree, size: "xs" })),
        React.createElement(Dropdown, { "data-testid": "breadcrumb-dropdown" }, items === null || items === void 0 ? void 0 : items.map(function (item) { return (React.createElement(DropdownItem, __assign({ label: item.label, key: item.label }, (typeof item.link === "string"
            ? { as: ChakraLink, href: item.link }
            : __assign({}, item.link))), item.label)); }))));
};
export var BreadcrumbDesktop = forwardRef(function (props, ref) {
    var items = props.items, showMaxItems = props.showMaxItems, variant = props.variant;
    var styles = useChakraMultiStyleConfig("Breadcrumb", {
        variant: variant
    });
    var containerProps = _omit(props, ["items", "showMaxItems", "variant"]);
    var getCustomList = function () {
        var hiddenItems = {
            as: HiddenItems,
            items: items.slice(2, -2)
        };
        var list = __spreadArray(__spreadArray(__spreadArray([], items.slice(0, 2), true), [
            { label: DEFAULT_MENU_BUTTON_LABEL, link: hiddenItems }
        ], false), items.slice(-2), true);
        return list;
    };
    var customList = showMaxItems ? getCustomList() : items;
    return (React.createElement(ChakraBreadcrumb, __assign({ "data-testid": "breadcrumb-desktop", ref: ref, separator: React.createElement(PhosphorIcon, { as: CaretRight }), spacing: 2, sx: styles.container }, containerProps), customList.map(function (item, index) { return (React.createElement(ChakraBreadcrumbItem, { isCurrentPage: customList.length === index + 1, key: item.label, sx: styles.item }, item.isLoading ? (React.createElement(ChakraSkeleton, { "data-testid": "breadcrumb-loading-item", minH: 2, minW: 20 })) : (React.createElement(ChakraBreadcrumbLink, __assign({}, (typeof item.link === "string"
        ? { href: item.link }
        : __assign({}, item.link)), { sx: styles.link }), item.label)))); })));
});
