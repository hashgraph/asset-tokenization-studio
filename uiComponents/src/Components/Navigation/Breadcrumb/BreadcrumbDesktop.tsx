import React from "react";
import {
  Breadcrumb as ChakraBreadcrumb,
  BreadcrumbItem as ChakraBreadcrumbItem,
  BreadcrumbLink as ChakraBreadcrumbLink,
  Link as ChakraLink,
  Menu,
  MenuButton,
  Skeleton as ChakraSkeleton,
} from "@chakra-ui/react";
import type { ComponentWithAs } from "@chakra-ui/system";
import {
  forwardRef,
  useMultiStyleConfig as useChakraMultiStyleConfig,
} from "@chakra-ui/system";
import { PhosphorIcon } from "../../Foundations/PhosphorIcon";
import { omit as _omit } from "lodash";
import { CaretRight, DotsThree } from "@phosphor-icons/react";
import type { BreadcrumbProps } from "./Breadcrumb";
import { Dropdown, DropdownItem } from "@Components/DataDisplay/Dropdown";

const DEFAULT_MENU_BUTTON_LABEL = "...";

const HiddenItems = ({ items }: { items: BreadcrumbProps["items"] }) => {
  const { menu } = useChakraMultiStyleConfig("Breadcrumb");

  return (
    <Menu data-testid="breadcrumb-menu">
      <MenuButton data-testid="breadcrumb-menu-button" sx={menu}>
        <PhosphorIcon as={DotsThree} size="xs" />
      </MenuButton>
      <Dropdown data-testid="breadcrumb-dropdown">
        {items?.map((item) => (
          <DropdownItem
            label={item.label}
            key={item.label}
            {...(typeof item.link === "string"
              ? { as: ChakraLink, href: item.link }
              : { ...item.link })}
          >
            {item.label}
          </DropdownItem>
        ))}
      </Dropdown>
    </Menu>
  );
};

export const BreadcrumbDesktop: ComponentWithAs<"div", BreadcrumbProps> =
  forwardRef<BreadcrumbProps, "div">((props: BreadcrumbProps, ref) => {
    const { items, showMaxItems, variant } = props;
    const styles = useChakraMultiStyleConfig("Breadcrumb", {
      variant,
    });
    const containerProps = _omit(props, ["items", "showMaxItems", "variant"]);

    const getCustomList = () => {
      const hiddenItems = {
        as: HiddenItems,
        items: items.slice(2, -2),
      };

      const list = [
        ...items.slice(0, 2),
        { label: DEFAULT_MENU_BUTTON_LABEL, link: hiddenItems },
        ...items.slice(-2),
      ];

      return list;
    };

    const customList = showMaxItems ? getCustomList() : items;

    return (
      <ChakraBreadcrumb
        data-testid="breadcrumb-desktop"
        ref={ref}
        separator={<PhosphorIcon as={CaretRight} />}
        spacing={2}
        sx={styles.container}
        {...containerProps}
      >
        {customList.map((item, index) => (
          <ChakraBreadcrumbItem
            isCurrentPage={customList.length === index + 1}
            key={item.label}
            sx={styles.item}
          >
            {item.isLoading ? (
              <ChakraSkeleton
                data-testid="breadcrumb-loading-item"
                minH={2}
                minW={20}
              />
            ) : (
              <ChakraBreadcrumbLink
                {...(typeof item.link === "string"
                  ? { href: item.link }
                  : { ...item.link })}
                sx={styles.link}
              >
                {item.label}
              </ChakraBreadcrumbLink>
            )}
          </ChakraBreadcrumbItem>
        ))}
      </ChakraBreadcrumb>
    );
  });
