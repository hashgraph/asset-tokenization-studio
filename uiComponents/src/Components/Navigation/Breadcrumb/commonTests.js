var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import React from "react";
export var defaultProps = {
    items: [
        {
            label: "Level 0",
            link: "/"
        },
        {
            label: "Level 1",
            link: "/level1"
        },
        {
            label: "Level 2",
            link: "/level1/level2"
        },
        {
            label: "Level 3",
            link: "/level1/level2/level3"
        },
        {
            label: "Level 4",
            link: "/level1/level2/level3/level4"
        },
        {
            label: "Level 5",
            link: "/level1/level2/level3/level4/level5"
        },
        {
            label: "Level 6",
            link: "http://www.google.com"
        },
        {
            label: "Level 7",
            link: "/level1/level2/level3/level4/level5/level6/level7"
        },
        {
            label: "Level 8",
            link: null
        },
    ],
    showMaxItems: false
};
export var CustomLink = function (_a) {
    var children = _a.children, props = __rest(_a, ["children"]);
    return (React.createElement("a", __assign({}, props, { style: { color: "blue", fontSize: "24px" } }), children));
};
export var customProps = {
    items: [
        {
            label: "Level 0",
            link: {
                as: CustomLink,
                "data-testid": "custom-link-0",
                href: "/level0"
            }
        },
        {
            label: "Level 1",
            link: {
                as: CustomLink,
                "data-testid": "custom-link-1",
                href: "/level1"
            }
        },
        {
            label: "Level 2",
            link: {
                as: CustomLink,
                "data-testid": "custom-link-2",
                href: "/level2"
            }
        },
        {
            label: "Level 3",
            link: {
                as: CustomLink,
                "data-testid": "custom-link-3",
                href: "/level3"
            }
        },
        {
            label: "Level 4",
            link: {
                as: CustomLink,
                "data-testid": "custom-link-4",
                href: "/level4"
            }
        },
        {
            label: "Level 5",
            link: {
                as: CustomLink,
                "data-testid": "custom-link-5",
                href: "/level5"
            }
        },
        {
            label: "Level 6",
            link: {
                as: CustomLink,
                "data-testid": "custom-link-6",
                href: "/level6"
            }
        },
        {
            label: "Level 7",
            link: {
                as: CustomLink,
                "data-testid": "custom-link-7",
                href: "http://www.google.com"
            }
        },
        {
            label: "Level 8",
            link: {
                as: CustomLink,
                "data-testid": "custom-link-8"
            }
        },
    ],
    showMaxItems: true
};
