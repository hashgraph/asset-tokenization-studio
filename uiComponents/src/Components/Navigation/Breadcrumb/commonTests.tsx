import React from "react";
import type { BreadcrumbProps } from "./Breadcrumb";

export const defaultProps: BreadcrumbProps = {
  items: [
    {
      label: "Level 0",
      link: "/",
    },
    {
      label: "Level 1",
      link: "/level1",
    },
    {
      label: "Level 2",
      link: "/level1/level2",
    },
    {
      label: "Level 3",
      link: "/level1/level2/level3",
    },
    {
      label: "Level 4",
      link: "/level1/level2/level3/level4",
    },
    {
      label: "Level 5",
      link: "/level1/level2/level3/level4/level5",
    },
    {
      label: "Level 6",
      link: "http://www.google.com",
    },
    {
      label: "Level 7",
      link: "/level1/level2/level3/level4/level5/level6/level7",
    },
    {
      label: "Level 8",
      link: null,
    },
  ],
  showMaxItems: false,
};

export const CustomLink = ({
  children,
  ...props
}: {
  children: React.ReactNode;
}) => {
  return (
    <a {...props} style={{ color: "blue", fontSize: "24px" }}>
      {children}
    </a>
  );
};

export const customProps: BreadcrumbProps = {
  items: [
    {
      label: "Level 0",
      link: {
        as: CustomLink,
        "data-testid": "custom-link-0",
        href: "/level0",
      },
    },
    {
      label: "Level 1",
      link: {
        as: CustomLink,
        "data-testid": "custom-link-1",
        href: "/level1",
      },
    },
    {
      label: "Level 2",
      link: {
        as: CustomLink,
        "data-testid": "custom-link-2",
        href: "/level2",
      },
    },
    {
      label: "Level 3",
      link: {
        as: CustomLink,
        "data-testid": "custom-link-3",
        href: "/level3",
      },
    },
    {
      label: "Level 4",
      link: {
        as: CustomLink,
        "data-testid": "custom-link-4",
        href: "/level4",
      },
    },
    {
      label: "Level 5",
      link: {
        as: CustomLink,
        "data-testid": "custom-link-5",
        href: "/level5",
      },
    },
    {
      label: "Level 6",
      link: {
        as: CustomLink,
        "data-testid": "custom-link-6",
        href: "/level6",
      },
    },
    {
      label: "Level 7",
      link: {
        as: CustomLink,
        "data-testid": "custom-link-7",
        href: "http://www.google.com",
      },
    },
    {
      label: "Level 8",
      link: {
        as: CustomLink,
        "data-testid": "custom-link-8",
      },
    },
  ],
  showMaxItems: true,
};
