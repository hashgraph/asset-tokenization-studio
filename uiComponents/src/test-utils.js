import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";
var DefaultTheme = extendTheme(BasePlatformTheme);
export var render = function (ui, theme) {
    if (theme === void 0) { theme = DefaultTheme; }
    var wrapper = function (_a) {
        var children = _a.children;
        return (React.createElement(ChakraProvider, { theme: theme }, children));
    };
    return testingLibraryRender(ui, { wrapper: wrapper });
};
