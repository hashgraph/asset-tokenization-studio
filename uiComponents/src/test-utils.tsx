import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import type { RenderResult } from "@testing-library/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";

const DefaultTheme = extendTheme(BasePlatformTheme);

export const render = (ui: JSX.Element, theme = DefaultTheme): RenderResult => {
  const wrapper = ({ children }: { children?: React.ReactNode }) => (
    <ChakraProvider theme={theme}>{children}</ChakraProvider>
  );

  return testingLibraryRender(ui, { wrapper });
};
