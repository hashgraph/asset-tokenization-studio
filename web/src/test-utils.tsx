import React from "react";
import { ChakraProvider } from "@chakra-ui/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import type { RenderOptions, RenderResult } from "@testing-library/react";
import { render, waitFor } from "@testing-library/react";
import type { MemoryHistory } from "history";
import { createMemoryHistory } from "history";
import { I18nextProvider } from "react-i18next";
import { Router } from "react-router-dom";
import i18n from "./i18n";
import theme from "./theme";
import userEvent from "@testing-library/user-event";

export const selectCalendar = async (
  component: RenderResult,
  id: string,
  day: string | number = new Date().getDate(),
) => {
  const calendar = component.getByTestId(id);
  await userEvent.click(calendar);

  await waitFor(() => {
    const daysToSelect = component.getAllByTestId(`day-${day}`);

    const dayToSelect = daysToSelect.find(
      (day) => !day.hasAttribute("disabled"),
    );

    if (dayToSelect) {
      userEvent.click(dayToSelect);
    }
  });

  await userEvent.click(document.body);
};

export const testQueryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      refetchOnMount: true,
      cacheTime: 0,
    },
  },
  logger: {
    log: console.log,
    warn: console.warn,
    // âœ… no more errors on the console for tests
    error: () => {},
  },
});

export const queryWrapper = ({ children }: { children: React.ReactNode }) => (
  <QueryClientProvider client={testQueryClient}>{children}</QueryClientProvider>
);

const memoryHistory = createMemoryHistory();

export const AllProviders = ({
  children,
  history = memoryHistory,
}: {
  children?: React.ReactNode;
  history?: MemoryHistory;
}) => {
  return (
    <I18nextProvider i18n={i18n}>
      <QueryClientProvider client={testQueryClient}>
        <ChakraProvider theme={theme}>
          <Router location={history.location} navigator={history}>
            {children}
          </Router>
        </ChakraProvider>
      </QueryClientProvider>
    </I18nextProvider>
  );
};

const customRender = (
  ui: React.ReactElement,
  {
    options,
    history,
  }: {
    options?: RenderOptions;
    history?: MemoryHistory;
  } = {},
): RenderResult =>
  render(ui, {
    wrapper: ({ children }) => (
      <AllProviders history={history}>{children}</AllProviders>
    ),
    ...options,
  });

export { customRender as render };
