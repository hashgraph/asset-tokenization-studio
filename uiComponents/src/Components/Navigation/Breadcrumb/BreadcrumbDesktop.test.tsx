import React from "react";
import { render } from "@/test-utils";
import { waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import type { BreadcrumbItemProps, BreadcrumbProps } from "./Breadcrumb";
import { BreadcrumbDesktop } from "./BreadcrumbDesktop";
import { customProps, defaultProps } from "./commonTests";

describe(`<BreadcrumbDesktop />`, () => {
  const factoryComponent = (props: BreadcrumbProps) =>
    render(<BreadcrumbDesktop {...props} />);

  test("renders correctly default", () => {
    const component = factoryComponent(defaultProps);

    expect(component.asFragment()).toMatchSnapshot("default");
  });

  test("renders correctly with maxItems to show", () => {
    const component = factoryComponent({ ...defaultProps, showMaxItems: true });

    expect(component.asFragment()).toMatchSnapshot("showMaxItems");
  });

  test("if showMaxItems is true should show a button to open hidden elements", async () => {
    const component = factoryComponent({ ...defaultProps, showMaxItems: true });

    const menuList = component.getByTestId("breadcrumb-dropdown");
    expect(menuList).toBeInTheDocument();
    expect(menuList).not.toBeVisible();

    const hiddenElement = component.getByText("Level 2");
    expect(hiddenElement).toBeInTheDocument();
    expect(hiddenElement).not.toBeVisible();

    const menuButton = component.getByTestId("breadcrumb-menu-button");
    expect(menuButton).toBeInTheDocument();

    await waitFor(() => {
      userEvent.click(menuButton);
      expect(menuList).toBeVisible();
      expect(hiddenElement).toBeVisible();
    });
  });

  test("should have a link for each element", async () => {
    const component = factoryComponent(defaultProps);

    const breadcrumb = component.getByTestId("breadcrumb-desktop");
    const elements = breadcrumb.getElementsByClassName(
      "chakra-breadcrumb__link"
    );

    expect(elements.length).toEqual(defaultProps.items.length);
  });

  test("renders custom links", () => {
    const component = factoryComponent(customProps);

    const { items } = customProps;

    items.forEach((item: BreadcrumbItemProps, index: number) => {
      const link = component.getByTestId(`custom-link-${index}`);
      expect(link).toBeInTheDocument();
    });
  });

  test("renders loading item", () => {
    const component = factoryComponent({
      ...defaultProps,
      items: [
        ...defaultProps.items,
        {
          label: "Loading",
          link: "/loading",
          isLoading: true,
        },
      ],
    });

    const loadingItem = component.getByTestId("breadcrumb-loading-item");
    expect(loadingItem).toBeInTheDocument();
  });
});
