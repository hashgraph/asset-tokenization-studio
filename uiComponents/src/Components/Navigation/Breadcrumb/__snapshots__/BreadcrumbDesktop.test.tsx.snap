// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BreadcrumbDesktop /> renders correctly default: default 1`] = `
<DocumentFragment>
  <nav
    aria-label="breadcrumb"
    class="chakra-breadcrumb css-1y2eexs"
    data-testid="breadcrumb-desktop"
  >
    <ol
      class="chakra-breadcrumb__list css-70qvj9"
    >
      <li
        class="chakra-breadcrumb__list-item css-brt312"
      >
        <a
          class="chakra-breadcrumb__link css-2y15zy"
          href="/"
        >
          Level 0
        </a>
        <span
          class="css-ouj2o6"
          role="presentation"
        >
          <svg
            class="chakra-icon css-kpbtv0"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
            />
          </svg>
        </span>
      </li>
      <li
        class="chakra-breadcrumb__list-item css-brt312"
      >
        <a
          class="chakra-breadcrumb__link css-2y15zy"
          href="/level1"
        >
          Level 1
        </a>
        <span
          class="css-ouj2o6"
          role="presentation"
        >
          <svg
            class="chakra-icon css-kpbtv0"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
            />
          </svg>
        </span>
      </li>
      <li
        class="chakra-breadcrumb__list-item css-brt312"
      >
        <a
          class="chakra-breadcrumb__link css-2y15zy"
          href="/level1/level2"
        >
          Level 2
        </a>
        <span
          class="css-ouj2o6"
          role="presentation"
        >
          <svg
            class="chakra-icon css-kpbtv0"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
            />
          </svg>
        </span>
      </li>
      <li
        class="chakra-breadcrumb__list-item css-brt312"
      >
        <a
          class="chakra-breadcrumb__link css-2y15zy"
          href="/level1/level2/level3"
        >
          Level 3
        </a>
        <span
          class="css-ouj2o6"
          role="presentation"
        >
          <svg
            class="chakra-icon css-kpbtv0"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
            />
          </svg>
        </span>
      </li>
      <li
        class="chakra-breadcrumb__list-item css-brt312"
      >
        <a
          class="chakra-breadcrumb__link css-2y15zy"
          href="/level1/level2/level3/level4"
        >
          Level 4
        </a>
        <span
          class="css-ouj2o6"
          role="presentation"
        >
          <svg
            class="chakra-icon css-kpbtv0"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
            />
          </svg>
        </span>
      </li>
      <li
        class="chakra-breadcrumb__list-item css-brt312"
      >
        <a
          class="chakra-breadcrumb__link css-2y15zy"
          href="/level1/level2/level3/level4/level5"
        >
          Level 5
        </a>
        <span
          class="css-ouj2o6"
          role="presentation"
        >
          <svg
            class="chakra-icon css-kpbtv0"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
            />
          </svg>
        </span>
      </li>
      <li
        class="chakra-breadcrumb__list-item css-brt312"
      >
        <a
          class="chakra-breadcrumb__link css-2y15zy"
          href="http://www.google.com"
        >
          Level 6
        </a>
        <span
          class="css-ouj2o6"
          role="presentation"
        >
          <svg
            class="chakra-icon css-kpbtv0"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
            />
          </svg>
        </span>
      </li>
      <li
        class="chakra-breadcrumb__list-item css-brt312"
      >
        <a
          class="chakra-breadcrumb__link css-2y15zy"
          href="/level1/level2/level3/level4/level5/level6/level7"
        >
          Level 7
        </a>
        <span
          class="css-ouj2o6"
          role="presentation"
        >
          <svg
            class="chakra-icon css-kpbtv0"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
            />
          </svg>
        </span>
      </li>
      <li
        class="chakra-breadcrumb__list-item css-brt312"
      >
        <span
          aria-current="page"
          class="chakra-breadcrumb__link css-2y15zy"
        >
          Level 8
        </span>
      </li>
    </ol>
  </nav>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<BreadcrumbDesktop /> renders correctly with maxItems to show: showMaxItems 1`] = `
<DocumentFragment>
  <nav
    aria-label="breadcrumb"
    class="chakra-breadcrumb css-1y2eexs"
    data-testid="breadcrumb-desktop"
  >
    <ol
      class="chakra-breadcrumb__list css-70qvj9"
    >
      <li
        class="chakra-breadcrumb__list-item css-brt312"
      >
        <a
          class="chakra-breadcrumb__link css-2y15zy"
          href="/"
        >
          Level 0
        </a>
        <span
          class="css-ouj2o6"
          role="presentation"
        >
          <svg
            class="chakra-icon css-kpbtv0"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
            />
          </svg>
        </span>
      </li>
      <li
        class="chakra-breadcrumb__list-item css-brt312"
      >
        <a
          class="chakra-breadcrumb__link css-2y15zy"
          href="/level1"
        >
          Level 1
        </a>
        <span
          class="css-ouj2o6"
          role="presentation"
        >
          <svg
            class="chakra-icon css-kpbtv0"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
            />
          </svg>
        </span>
      </li>
      <li
        class="chakra-breadcrumb__list-item css-brt312"
      >
        <button
          aria-controls="menu-list-:r1:"
          aria-expanded="false"
          aria-haspopup="menu"
          class="chakra-menu__menu-button css-12005wt"
          data-testid="breadcrumb-menu-button"
          id="menu-button-:r1:"
        >
          <span
            class="css-xl71ch"
          >
            <svg
              class="chakra-icon css-1qzovvq"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M140,128a12,12,0,1,1-12-12A12,12,0,0,1,140,128Zm56-12a12,12,0,1,0,12,12A12,12,0,0,0,196,116ZM60,116a12,12,0,1,0,12,12A12,12,0,0,0,60,116Z"
              />
            </svg>
          </span>
        </button>
        <div
          class="css-r6z5ec"
          style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
        >
          <div
            aria-orientation="vertical"
            class="chakra-menu__menu-list css-12srbko"
            data-testid="breadcrumb-dropdown"
            id="menu-list-:r1:"
            role="menu"
            style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
            tabindex="-1"
          >
            <div
              class="chakra-stack css-6ekxde"
            >
              <a
                class="chakra-link chakra-menu__menuitem css-1ow1hrd"
                data-index="0"
                data-testid="dropdown-item"
                href="/level1/level2"
                id="menu-list-:r1:-menuitem-:r2:"
                role="menuitem"
                tabindex="-1"
              >
                <p
                  class="css-1a1j0go"
                >
                  Level 2
                </p>
              </a>
              <a
                class="chakra-link chakra-menu__menuitem css-1ow1hrd"
                data-index="1"
                data-testid="dropdown-item"
                href="/level1/level2/level3"
                id="menu-list-:r1:-menuitem-:r3:"
                role="menuitem"
                tabindex="-1"
              >
                <p
                  class="css-1a1j0go"
                >
                  Level 3
                </p>
              </a>
              <a
                class="chakra-link chakra-menu__menuitem css-1ow1hrd"
                data-index="2"
                data-testid="dropdown-item"
                href="/level1/level2/level3/level4"
                id="menu-list-:r1:-menuitem-:r4:"
                role="menuitem"
                tabindex="-1"
              >
                <p
                  class="css-1a1j0go"
                >
                  Level 4
                </p>
              </a>
              <a
                class="chakra-link chakra-menu__menuitem css-1ow1hrd"
                data-index="3"
                data-testid="dropdown-item"
                href="/level1/level2/level3/level4/level5"
                id="menu-list-:r1:-menuitem-:r5:"
                role="menuitem"
                tabindex="-1"
              >
                <p
                  class="css-1a1j0go"
                >
                  Level 5
                </p>
              </a>
              <a
                class="chakra-link chakra-menu__menuitem css-1ow1hrd"
                data-index="4"
                data-testid="dropdown-item"
                href="http://www.google.com"
                id="menu-list-:r1:-menuitem-:r6:"
                role="menuitem"
                tabindex="-1"
              >
                <p
                  class="css-1a1j0go"
                >
                  Level 6
                </p>
              </a>
            </div>
          </div>
        </div>
        <span
          class="css-ouj2o6"
          role="presentation"
        >
          <svg
            class="chakra-icon css-kpbtv0"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
            />
          </svg>
        </span>
      </li>
      <li
        class="chakra-breadcrumb__list-item css-brt312"
      >
        <a
          class="chakra-breadcrumb__link css-2y15zy"
          href="/level1/level2/level3/level4/level5/level6/level7"
        >
          Level 7
        </a>
        <span
          class="css-ouj2o6"
          role="presentation"
        >
          <svg
            class="chakra-icon css-kpbtv0"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
            />
          </svg>
        </span>
      </li>
      <li
        class="chakra-breadcrumb__list-item css-brt312"
      >
        <span
          aria-current="page"
          class="chakra-breadcrumb__link css-2y15zy"
        >
          Level 8
        </span>
      </li>
    </ol>
  </nav>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
